lint target no longer works without $GOPATH

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!
If the matter is security sensitive, please disclose it privately to a security contact: https://github.com/kudobuilder/kudo/blob/master/SECURITY_CONTACTS
-->


**What happened**:

Checked out `master` and ran:

```
$ LANG=C make lint
if [ ! -f /bin/golangci-lint ]; then ./hack/install-golangcilint.sh; fi
golangci/golangci-lint info checking GitHub for tag 'v1.23.7'
golangci/golangci-lint info found version: 1.23.7 for v1.23.7/linux/amd64
golangci/golangci-lint info installed /home/porridge/go/bin/golangci-lint
/bin/golangci-lint run
/bin/bash: /bin/golangci-lint: No such file or directory
make: *** [Makefile:51: lint] Error 127
```

**What you expected to happen**:

Lint should pass on a clean repo.

**How to reproduce it (as minimally and precisely as possible)**:

`make lint`

**Anything else we need to know?**:

- I don't have `GOPATH` set, which might be related.
- I think there was a recent bump of the linter cc @zen-dog

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -34,7 +34,7 @@ You can find the full text of the DCO here: https://developercertificate.org/
 ### Pre-requisites
 
 - Git
-- Go `1.13` or later
+- Go `1.13` or later. Note that some [Makefile](Makefile) targets assume that your `$GOBIN` is in your `$PATH`.
 - [Kubebuilder](https://book.kubebuilder.io/quick-start.html#installation) version 2 or later - note that it is only needed for the `kube-apiserver` and `etcd` binaries, so no need to install *its* dependencies (such as `kustomize`).
 - A Kubernetes Cluster running version `1.13` or later (e.g., [kind](https://kind.sigs.k8s.io/) or [Minikube](https://kubernetes.io/docs/tasks/tools/install-minikube/))
 - [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)

--- a/Makefile
+++ b/Makefile
@@ -47,8 +47,10 @@ test-clean:
 
 .PHONY: lint
 lint:
-	if [ ! -f ${GOPATH}/bin/golangci-lint ]; then ./hack/install-golangcilint.sh; fi
-	${GOPATH}/bin/golangci-lint run
+ifeq (, $(shell which golangci-lint))
+	./hack/install-golangcilint.sh
+endif
+	golangci-lint run
 
 .PHONY: download
 download:

